### PR TITLE
Add more options for better control on CI

### PR DESCRIPTION
**Pull request by mattleibow**
Add 2 new flags to `avd create`:

* `--skin` for a nice look
* `--abi` to control which ABI is selected when there are multiple for a single image.

Add several flags for `avd start`:

* `--no-snapshot-load`
* `--no-snapshot-save`
* `--camera-back`
* `--camera-front`
* `--port` to control ports for predictability on CI
* `--screen` for controlling touch support
* `--engine` 
* `--accel-mode`
* `--no-accel`
* `--no-boot-anim` for faster boots
* `--no-window` for headless scenarios
* `--no-audio` for some CI machines with missing libraries
* `--gpu` to pick a different system on nested hypervisors on CI
* `--no-jni`
* `--verbose` for extra details

### CODE REVIEW

File: .github/workflows/run.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Run Android Emulator
+
+on:
+  pull_request:
+  push:
+
+jobs:
```
Comment (mattleibow) [inline .github/workflows/run.yml:7]: TODO: this just ignores the return codes of the commands so all fails are hidden.

File: AndroidSdk/AvdManager/AvdManager.cs [hunk 2]
```diff
@@ -59,45 +60,64 @@ public override FileInfo FindToolPath(DirectoryInfo androidSdkHome)
...
 
-			run(args.ToArray());
+			run(args.ToArray(), input =>
+			{
+				// Continuously send "ENTER" to respond to the prompt:
+				// "Do you wish to create a custom hardware profile? [no]"
+				// Sending "no" too soon will cause an error...
+				input.WriteLine();
+				input.Flush();
+			});
```
Comment (mattleibow) [inline AndroidSdk/AvdManager/AvdManager.cs:115]: This happens in some situations when you are on a new machine and there is no "hardware profile"... Not sure what that means, but it usually is on CI.

File: AndroidSdk/AvdManager/AvdManager.cs [hunk 3]
```diff
@@ -279,7 +299,7 @@ void ParseIniFile(Dictionary<string, string> properties, string iniFile)
 			}
 		}
 
-		IEnumerable<string> run(params string[] args)
+		IEnumerable<string> run(string[] args, Action<StreamWriter> inputAction = null)
```
Comment (mattleibow) [inline AndroidSdk/AvdManager/AvdManager.cs:302]: I notice this code is basically duplicated 3 times in each tool, so I want to try and refactor that into a separate runner thing that we can just reuse - especially with the "keep posting input until we are done" 10x engineering.

Comment (mattleibow) [reply AndroidSdk/AvdManager/AvdManager.cs]: I see this is already done via the new ProcessRunner used in adbd. So will just update the old tools in a new PR.

File: .github/workflows/run.yml (hunk 1 above)
Comment (mattleibow) [reply .github/workflows/run.yml:7]: I'll fix this in a new PR.

**Review by Redth — APPROVED** [review]

**Pull request merged**